### ISSUE TITLE
Scripts fail with "Permission denied to access object"

### ISSUE THREAD

**Issue by Sollace** [violentmonkey/violentmonkey#399]
This just suddenly started happening with all my scripts (except one that checks for a public variable to see if it's on the right page).

```
one
two
four
Error running script: Tets
Error: Permission denied to access object
Permission denied to access object
```

I managed to reproduce it with this snippet:

```js
// ==UserScript==
// @name Tets
// @namespace fimf
// @include     /^http?[s]://www.fimfiction.net/.*/
// @grant       none
// @run-at      document-start
// ==/UserScript==

console.log("one");
new Array();
console.log("two");
document.addEventListener('click', a => a);
console.log("three");
const colours = {Mapping: {}, Keys: [], Names: [], NamesLower: [], Sets: {
    'Standard Colours': [0, [0,73,70,68,107,59,103,24,112]],
    'FimFiction': [0, range(113,127)],
    'Mane Six': [0, range(128, 133)],
    'More Colours': [1, range(0,112)] }
};
console.log("four");

function arrayOf(length, func) {return Array.apply(null, Array(length)).map(func);}
function range(from, to) {return arrayOf(to - from + 1, (_, i) => from + i);}
```

Something has changed that prevents scripts from accessing globals, globals from accessing scripts, or _something_. I've also noticed my other scripts have become sandboxed, which breaks them in other ways.

Browser: `Mozilla Firefox 59.0.2`
Addon: `Violentmonkeyv2.9.2`

**Comment by Sollace** [comment]
Update:

This also doesn't work.

```js
console.log("one");
Object.keys({});
console.log("two");
```

**Comment by Sollace** [comment]
Wew, been ages since I came here. I'm going to add a little more information:

The problem seems to stem from the fix for #385. There's probably more that has to be done to get sandboxing to function correctly. I know _I_ suggested the change, but I think the ideal approach would be to only use it on pages where it's required (ones that have a CSP that would normally block script execution).

**Comment by gera2ld** [comment]
After a few experiments, I think it's better not to use `tabs.executeScript` because all scripts run in the same sandbox, and they are actually not able to access `unsafeWindow` even though they have a reference of `wrappedJSObject`. Some more details:
- In the sandbox, all scripts share the same context with the content script, with more permissions than they should have.
- Many scripts want to access to properties of `unsafeWindow` without limitations, which is impossible if the scripts run in the sandbox. Possible result is `Permission denied to access object`.
- Many scripts modify prototypes of global objects such as `Array` and `Object`. Though it's not recommended, I don't think we can just drop support for them. When running from a sandbox, the scripts will have different global objects. As a result, the modifications may not work as expected.
- `wrappedJSObject` is for Firefox only. And it is not that great.

So I don't think we will use such workaround in the future. Please confirm if your issue still happen in the latest v2.9.4.
